### ISSUE TITLE
Release 3.3.0 beta - Receiving "ERROR:  This is only compatible with Python 3. Please install Python3" on startup

### ISSUE THREAD

**Issue by mjarends** [greghesp/assistant-relay#162]
**Describe the bug**
I am receiving the following error message when starting the 3.3.0 beta release after a fresh install

> ERROR:  This is only compatible with Python 3. Please install Python3

**To Reproduce**
1. Stopped PM2 processes
2. Backed up conifg.json and audio-responses folders
3. Removed previous release of Google Assistant Relay
4. Downloaded 3.3.0 beta release and extracted it to my system
5. Ran npm install from the install directory
5. Put the config.json and audio-responses back
6. Started Google assistant relay

**Expected behavior**
Expected the application to start successfully

**Screenshots**
N/A

**Desktop (please complete the following information):**
-  Platform: Windows 10 Professional
 - Node.js Version: 12.13.1
 - GAR Version: 3.3.0b

**Additional context**

**Comment by ryancasler** [comment]
Do you have python installed as well?

**Comment by mjarends** [comment]
Yes. I have both Python 2.7 and 3.x installed. Neither are available on the system path though.

**Comment by ryancasler** [comment]
I'm sorry but I don't know what that means.  I can tell you that GAR uses the exec function in NodeJS to execute the python code for CATT.  So, if NodeJS is unable to run the python code, then that would explain why you're not able to run it.

**Comment by glocklol** [comment]
Getting the same issue here on CentOS 7.7 with py2.7 & py3.7 installed.

**Comment by greghesp** [comment]
Can you try running pip3 in the terminal? 
You need pip3 installed, which is what it checks for.

**Comment by glocklol** [comment]
Well that seems to have done it. Though I'm still getting complaints from pm2 about this package only being compatible with python3:

2|www      | ERROR:  This is only compatible with Python 3. Please install Python3

**Comment by greghesp** [comment]
Is that an old log, pm2 keeps them

It's checking to see if pip3 is a valid terminal command. If it isn't, it errors

`if(!s.which('pip3'))`

**Comment by mjarends** [comment]
I'm away from home right now but will try to test the same when I'm back in a couple of days.

**Comment by matthewnrauch** [comment]
PIP3 was missing from my Raspberry Pi 3 B+ setup.  I had this same error, but after installing that I was able to get it to setup and run without any issues.  I would suggest you add PIP3 to the requirements so others don't run into this issue.  Especially with a Raspberry Pi.

**Comment by mjarends** [comment]
Sorry it's taken me so long to respond.

The issue is that I run this on Windows 10 and I do not add either Python 2.7 or 3.8 to my PATH environment variable. Instead I rely on the Windows Python Launcher to determine which Python version to run. See https://docs.python.org/3/using/windows.html#python-launcher-for-windows for more details.

Pip is still callable using the Windows Python launcher but you need to call it using:

`py -3 -m pip`

If you want to install a module:

`py -3 -m pip install my_package`

Run a script:

`py -3 my_script.py`

**Comment by greghesp** [comment]
Ok, similar to me.
I think I'm good to have to revist how it installs the dependencies. I'll add an install option to the dashboard where users can specify the command to run python

**Comment by greghesp** [comment]
@mjarends Can you try the [latest beta release](https://github.com/greghesp/assistant-relay/releases/tag/v3.3.2b)

See if that works?

**Comment by mjarends** [comment]
@greghesp Just downloaded and tried to enable casting.

I entered the following for the pip3 command: 
`py -3 -m pip`

And received the following error:
`This is only compatible with Python 3. Please install Python3 and ensure pip3 is available from your terminal`

Another thing that I noticed is that the pip3 Command field keeps losing focus when I type.

I was wondering if you could implement all of this logic within your code instead of through a config property. I was thinking you could first to see which platform you are on. If on Windows then see if the py.exe is available on the path. If it is call pip using `py -3 -m pip`, otherwise revert to using `pip3` directly.

**Comment by greghesp** [comment]
> Another thing that I noticed is that the pip3 Command field keeps losing focus when I type.

That shouldn't be happening, and its probably the issue.  I'll look into that.

In the mean time, in your `/bin/conig.json` file, can you tell me what the "pipCommand" field is set to?

**Comment by mjarends** [comment]
I have the following in config.json

```
  "castEnabled": true,
  "pipCommand": "py -3 -m pip"
```

**Comment by greghesp** [comment]
I can't replicate your issue at all @mjarends 

The code 
```
        const db = await low(adapter);
        const pip = db.get('pipCommand').value();

        if(!s.which(pip)){
            return rej("This is only compatible with Python 3. Please install Python3 and ensure pip3 is available from your terminal");
        }
```

takes the value from the config and executes it.  I've been able to get this to work and fail on my system so I know its working.  Just not sure why it wouldn't work for you :/

**Comment by greghesp** [comment]
I've just realised whats wrong here.  Completely missed that you don't have this added to the path.

@mjarends  For now, in your `/helpers/cast.js` file, comment out lines 16-18:
https://github.com/greghesp/assistant-relay/blob/master/relay/helpers/cast.js#L16-L18

This will stop the check on install.

**Comment by mjarends** [comment]
@greghesp You are correct.

I worked around the issue by modifying both install lines with ${pip}... and adding the full path to catt in the shelljs .exec() and spawn() calls. After doing that GAR loads correctly and I am able to send test cast requests.

Do you know if it's possible to pass environment variables into PM2? What I would like to do is pass the location of the Python3 Scripts directory in when the process starts.

**Comment by greghesp** [comment]
> Do you know if it's possible to pass environment variables into PM2? What I would like to do is pass the location of the Python3 Scripts directory in when the process starts.

Give this a go https://pm2.keymetrics.io/docs/usage/environment/

**Comment by mjarends** [comment]
I was able to get everything working. What I did was added the Python 3 Scripts directory to the Windows Path in the script that I use to launch GAR.

I believe that if I added both the Python 3 root and Scripts directory to the path it the pip install and catt calls would all work. If needed I can verify this.

It's up to you whether you want to leave the pip install command in the application. Thanks for the help you've given me.
